File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main {
    static boolean[] visited;
    static boolean[][] networkMap;
    static int result;

    public static void main(String[] args) {

        try (BufferedReader bf = new BufferedReader(new InputStreamReader(System.in))) {
            int computerCount = Integer.parseInt(bf.readLine());
            visited = new boolean[computerCount + 1];

            int network = Integer.parseInt(bf.readLine());

            networkMap = new boolean[computerCount + 1][computerCount + 1];

            for (int i = 0; i < network; i++) {
                String[] inputs = bf.readLine().split(" ");
                int num1 = Integer.parseInt(inputs[0]);
                int num2 = Integer.parseInt(inputs[1]);

                networkMap[num1][num2] = true;
                networkMap[num2][num1] = true;
            }

            DFS(1);

            System.out.println(result - 1);
        } catch (Exception e) {
            e.printStackTrace();
        }

    }

    static void DFS(int num) {
        visited[num] = true;
        result++;
        for (int i = 1; i < networkMap[num].length; i++) {
            if (networkMap[num][i] && !visited[i]) {
                DFS(i);
            }
        }
    }

}

File: 백준/Silver/2606. 바이러스/바이러스.js
const solution = (computerMap) => {
  visited = Array(computer + 1).fill(false);
  computers = new Array(computer + 1)
    .fill()
    .map(() => new Array(computer + 1).fill(false));

  computerMap.forEach((x) => {
    const [num1, num2] = x;
    computers[num1][num2] = true;
    computers[num2][num1] = true;
  });

  DFS(1);

  return result - 1;
};

const DFS = (num) => {
  visited[num] = true;
  result++;
  for (let i = 1; i < computers.length; i++) {
    if (visited[i] === false && computers[num][i]) {
      DFS(i);
    }
  }
};

const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let computer = 0;
let lines = 0;
const computerMap = [];
let result = 0;
let visited;
let computers;

rl.on("line", function (line) {
  if (computer === 0) {
    computer = +line;
  } else if (lines === 0) {
    lines = +line;
  } else {
    computerMap.push(line.split(" ").map(Number));
  }

  if (computer !== 0 && lines !== 0 && computerMap.length === lines) {
    rl.close();
  }
}).on("close", function () {
  console.log(solution(computerMap));
  process.exit();
});
